src/EFTools/EntityDesignerVersioningFacade/ReverseEngineerDb/SchemaDiscovery/EntityStoreSchemaQueryGenerator.cs

```diff
@@ -149,15 +149,24 @@ internal StringBuilder CreateWhereClause(EntityParameterCollection parameters)
 
             if (value != null)
             {
-                var parameterName = "p" + parameters.Count.ToString(CultureInfo.InvariantCulture);
+                var matchingParameter = parameters.GetParameterByValue(value);
```

Comment:
If we simplify the matching logic to just do one equality check then I think it makes sense to revisit @lajones's original suggestion to use a dictionary and some `GetOrAdd`-like logic here.